### PR TITLE
feat(indexer): Preallocate space for a slice

### PR DESCRIPTION
**Pull request by roninjin10**
Optimize mapping of db data to api response by preallocating a slice since we know it's length ahead of time

### CODE REVIEW

**Comment by roninjin10** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6738** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6738" target="_blank">Graphite</a> 
    * **PR #6767** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6767" target="_blank">Graphite</a>  👈
      * **PR #6773** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6773" target="_blank">Graphite</a> 
        * **PR #6768** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6768" target="_blank">Graphite</a> 
          * **PR #6726** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6726" target="_blank">Graphite</a> 
            * **PR #6799** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6799" target="_blank">Graphite</a> 
      * **PR #6771** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6771" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6767?utm_source=stack-comment).

File: indexer/api/handlers/deposits.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ type DepositResponse struct {
 
 // TODO this is original spec but maybe include the l2 block info too for the relayed tx
 func newDepositResponse(deposits []*database.L1BridgeDepositWithTransactionHashes) DepositResponse {
-	var items []DepositItem
+	items := make([]DepositItem, 0, len(deposits))
```
Comment (roninjin10) [inline indexer/api/handlers/deposits.go]: Put this in it's own pr because I'm a Go NOOB and didn't know if there were any problems with doing this or if this is a good idea

Comment (hamdiallam) [reply indexer/api/handlers/deposits.go]: the second argument isn't needed
`make([]DepositItem, len(deposits))` is enough

**Review by hamdiallam — APPROVED** [review]

File: indexer/api/routes/deposits.go [hunk 2]
```diff
@@ -30,7 +30,7 @@ type DepositResponse struct {
 // TODO this is original spec but maybe include the l2 block info too for the relayed tx
 // FIXME make a pure function that returns a struct instead of newWithdrawalResponse
 func newDepositResponse(deposits []*database.L1BridgeDepositWithTransactionHashes) DepositResponse {
-	var items []DepositItem
+	items := make([]DepositItem, 0, len(deposits))
```
Comment (roninjin10) [inline indexer/api/routes/deposits.go]:
```suggestion
	items := make([]DepositItem, len(deposits))
```

File: indexer/api/routes/withdrawals.go [hunk 3]
```diff
@@ -43,7 +43,7 @@ type WithdrawalResponse struct {
 
 // FIXME make a pure function that returns a struct instead of newWithdrawalResponse
 func newWithdrawalResponse(withdrawals []*database.L2BridgeWithdrawalWithTransactionHashes) WithdrawalResponse {
-	var items []WithdrawalItem
+	items := make([]WithdrawalItem, 0, len(withdrawals))
```
Comment (roninjin10) [inline indexer/api/routes/withdrawals.go]:
```suggestion
	items := make([]WithdrawalItem, len(withdrawals))
```
